data-model/src/main/resources/META-INF/native-image/io.micronaut.data/data-model/native-image.properties

```diff
@@ -14,4 +14,8 @@
 # limitations under the License.
 #
 
-Args = --initialize-at-build-time=io.micronaut.data.model.query.builder.sql.Dialect,io.micronaut.data.model.DataType,io.micronaut.data.annotation.Join$Type
+Args = --initialize-at-build-time=io.micronaut.data.model.query.builder.sql.Dialect \
+  --initialize-at-build-time=io.micronaut.data.model.DataType \
+  --initialize-at-build-time=io.micronaut.data.annotation.Join$Type \
+  --initialize-at-build-time=io.micronaut.data.model.JsonDataType \
+  --initialize-at-build-time=io.micronaut.data.intercept.annotation.DataMethod$OperationType
```

Comment:
these flags shouldn't be needed if the project upgrades to Micronaut core 4.6.5

Reply:
@sdelamo Can you try it with the latest Core?

Reply:
With latest core and platform, looks like these are not needed for graalvm 21 (still failing for graalvm 17).